system/jlib/jsocket.cpp

```diff
@@ -6428,7 +6442,15 @@ class CSocketConnectWait: implements ISocketConnectWait, public CInterface
                 else
                 {   // select/poll failed
                     err = ERRNO();
-                    LOGERR(err,2,"CSocketConnectWait ::select/poll");
+                    if (err != JSE_INTR)
+                    {
+                        LOGERR(err,2,"CSocketConnectWait ::select/poll");
+                        sock->errclose();
+                        isopen = false;
+                        if (!oneshot)
+                            connectimedout = true; // to force same behavior as before
+                        break;
```

Comment:
Is it right to unconditional break out here?
The other if oneshot is false code looks like it's designed to handle the error and try again (as long as within the timeout period).

Reply:
Debatable, but we poll()ed for timeout duration and if poll() failed with something other than EINTR then hard to see why to keep trying until timeout.

Reply:
right, there's no way ::select might have exited before the timeout given to it?

And if not, isn't this just as true (if not more so) when oneshot==true, i.e. shouldn't it set connectimedout = true above either way?

I think the comment '// to force same behavior as before' should be removed or change. It won't be helpful to anyone reading the code. Could instead comment why that is' existing because it must be here because waited 'timeout' in ::select above.